### PR TITLE
Issue#127 Handle edge conditions for IDs

### PR DESCRIPTION
**Pull request by cragun47**
Improvements to handle edge conditions for ID's:

If the authority has blanks in the name, replace blanks with _'s so the System ID is valid
If the authority is not present in the HL7 message ID's, do not create a System ID in the ID resource
Improve SYSTEM_ID data handler to return null on input of null or empty to support no System ID creation.
Do not assume an authority for an ID created from a PID.19 SSN.

### CODE REVIEW

**Comment by cragun47** [comment]
Added to this PR:
    1.  Additional requested tests for email
    2. Cause calculated text from coding to not be shown for IDs.   ("SS", "MR", "NH", "DL", etc.)

File: src/main/resources/hl7/datatype/Identifier_SystemID.yml [hunk 1]
```diff
@@ -9,11 +9,17 @@ type:
    valueOf: datatype/CodeableConcept
    expressionType: resource
    specs: CX.5 | $type
+   constants: 
+      # Because calculated system id's don't have text, force the calculated text from code to be hidden
+      hideText: 'HIDE'
```
Comment (klwhaley) [inline src/main/resources/hl7/datatype/Identifier_SystemID.yml:14]: i need to think a bit more about this design to 'hide' the text...but will approve.  We may want to do more on this / change how we have done this...

File: src/main/resources/hl7/datatype/CodeableConcept.yml [hunk 2]
```diff
@@ -43,7 +43,10 @@ coding_4:
        display: CWE.5
        version: CWE.8
 
-text:  
+text:
+     # If hideText is not passed by parent, the value is NULL, making it backward compatible with all other uses of CodeableConcept
```
Comment (klwhaley) [inline src/main/resources/hl7/datatype/CodeableConcept.yml:47]: i need to think a bit more about this design to 'hide' the text...but will approve. We may want to do more on this / change how we have done this...

Comment (cragun47) [reply src/main/resources/hl7/datatype/CodeableConcept.yml:47]: Acknowledged.

**Pull request merged**